tests/test_numpy_array.cpp

```diff
@@ -528,20 +528,69 @@ TEST_SUBMODULE(numpy_array, sm) {
                return sum_str_values;
            });
 
+    sm.def("pass_array_handle_return_sum_str_values", [](const py::array_t<py::handle> &objs) {
```

Comment:
To avoid the code duplication, could you please try moving the existing code (lines 522 to 546) to a template helper function, then instantiate the helper function with `PyObject *`, `handle`, `object` (I'd call the template parameter `PyObjectType`.)

Reply:
Done